File: 其他功能/OC版--TbaleView索引高仿/OC版--TbaleView索引高仿/MGIndexView/Model/MGCarGroup.m
//
//  MGCarGroup.m
//  OC版--TbaleView索引高仿
//
//  Created by i-Techsys.com on 17/4/6.
//  Copyright © 2017年 i-Techsys. All rights reserved.
//

#import "MGCarGroup.h"
#import "MGCar.h"

@implementation MGCarGroup

+ (instancetype)groupWithDict:(NSDictionary *)dict
{
    return [[self alloc] initWithDict:dict];
}

- (instancetype)initWithDict:(NSDictionary *)dict
{
    if (self = [super init]) {
        // 赋值标题
        self.title = dict[@"title"];
        
        //        [self setValue:_title forKey:@"title"];
        
        // 取出原来的字典数组
        NSArray *dictArray = dict[@"cars"];
        NSMutableArray *carArray = [NSMutableArray array];
        for (NSDictionary *dict in dictArray) {
            MGCar *car = [MGCar carWithDict:dict];
            [carArray addObject:car];
        }
        self.cars = carArray;
    }
    return self;
}

@end

File: indexView/indexView/索引条/Model/MGCarGroup.swift
//
//  MGCarGroup.swift
//  indexView
//
//  Created by i-Techsys.com on 17/3/13.
//  Copyright © 2017年 i-Techsys. All rights reserved.
//

import UIKit

class MGCarGroup: NSObject {
    /**
     *  这组的标题
     */
    var title: String!
    /**
     *  存放的所有的汽车品牌(里面装的都是MJCar模型)
     */
    var cars: [[String : NSObject]]? {
        didSet {
            guard let cars = cars else { return }
            for dict in cars {
                carArrs.append(MGCar(dict: dict))
            }
        }
    }
    lazy var carArrs: [MGCar] = [MGCar]()
    
    override init() {
        
    }
    
    init(dict: [String: Any]) {
        super.init()
        setValuesForKeys(dict)
    }
    
    override func setValue(_ value: Any?, forUndefinedKey key: String) {
    }
}
